Tests/UnitTests/FileShareService.DesktopClientTests/Modules/Search/BatchDetailsViewModelTest.cs

```diff
@@ -0,0 +1,100 @@
+        private IMessageBoxService fakeMessageBoxService = null!;
+        private  IFileService fakeFileService = null!;
+        private IFileShareApiAdminClient fakeFileShareApiAdminClient = null!;
+        private ISaveFileDialogService fakesaveFileDialogService = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeFileShareApiAdminClientFactory = A.Fake<IFileShareApiAdminClientFactory>();
+            fakeMessageBoxService = A.Fake<IMessageBoxService>();
+            fakeFileService = A.Fake<IFileService>();
+            fakeFileShareApiAdminClient = A.Fake<IFileShareApiAdminClient>();
+            fakesaveFileDialogService = A.Fake<ISaveFileDialogService>();
+        }
+        [Test]
+        public async Task TestDownloadFileWithFileExistsAndYesResponse()
+        {
+           
+            var BatchDetailVM = new BatchDetailsViewModel(fakeFileShareApiAdminClientFactory, fakeMessageBoxService, fakeFileService , fakesaveFileDialogService);
+            
+            BatchDetailVM.Files = new List<BatchDetailsFiles>() { new BatchDetailsFiles() { Filename = "AFilename.txt", FileSize = 100 } };
+            A.CallTo(() => fakesaveFileDialogService.SaveFileDialog(A<string>.Ignored)).Returns(Path.Combine(Environment.CurrentDirectory));
+
+            BatchDetailVM.BatchId = Guid.NewGuid().ToString();
+            string fileDownloadPath = Path.Combine(Environment.CurrentDirectory, "AFilename.txt");          
+            A.CallTo(() => fakeFileService.Exists(fileDownloadPath)).Returns(true);
+            A.CallTo(() => fakeMessageBoxService.ShowMessageBox(A<string>.Ignored, A<string>.Ignored,MessageBoxButton.YesNo, A<MessageBoxImage>.Ignored)).Returns(MessageBoxResult.Yes);
+            A.CallTo(() => fakeFileShareApiAdminClientFactory.Build()).Returns(fakeFileShareApiAdminClient);
+            A.CallTo(() => fakeFileShareApiAdminClient.DownloadFileAsync(A<string>.Ignored, A<string>.Ignored, A<FileStream>.Ignored, A<long>.Ignored, A<CancellationToken>.Ignored)).Returns(new Result<DownloadFileResponse> { IsSuccess = true, StatusCode = 206 });
+
+            BatchDetailVM.DownloadExecutionCommand.Execute("AFilename.txt");
+
+            A.CallTo(() => fakeMessageBoxService.ShowMessageBox(A<string>.Ignored, A<string>.Ignored, MessageBoxButton.OK, A<MessageBoxImage>.Ignored)).MustHaveHappened();
+        } 
+
+        [Test]
+        public async Task TestDownloadFileWithFileExistsAndNoResponse()
+        {
+            var BatchDetailVM = new BatchDetailsViewModel(fakeFileShareApiAdminClientFactory, fakeMessageBoxService, fakeFileService, fakesaveFileDialogService);
+
+            BatchDetailVM.Files = new List<BatchDetailsFiles>() { new BatchDetailsFiles() { Filename = "AFilename.txt", FileSize = 100 } };
+            A.CallTo(() => fakesaveFileDialogService.SaveFileDialog(A<string>.Ignored)).Returns(Path.Combine(Environment.CurrentDirectory));
```

Comment:
Again please remove Path.Combine
```suggestion
            A.CallTo(() => fakesaveFileDialogService.SaveFileDialog(A<string>.Ignored)).Returns(Environment.CurrentDirectory);
```